### ISSUE TITLE
Statistics 

### ISSUE THREAD

**Issue by xxsimoxx** [codepotent/Update-Manager#18]
# Description
Have statistics about installed plugins.
I've already done it, but I want to keep it under tests for a few days before doing a PR.
If you want you can look at it [here](https://github.com/xxsimoxx/Update-Manager).

I've tried to keep it almost separate, so I've just added a constant and called the class in the main plugin file. The rest is handled by `classes/Statistics.class.html`.
My idea is to have something very simple.

Missing things:
- [ ] Hash the remote url (left in plain text just for better testing)
- [ ] Documentation
- [ ] GPDR notes
- [ ] Some variables left around that need discussion
- [ ] Discuss if populating the *active installations* filter 

Simone.

![image](https://user-images.githubusercontent.com/29772709/71418939-77b71780-266d-11ea-9e90-c54afd1e5e19.png)

**Comment by johnalarcon** [comment]
Hi Simone,

I think we've had a miscommunication. I'm not planning to integrate stats _into_ the plugin... 

The `codepotent_update_manager_filter_request` filter was added to allow 3rd party devs to capture, process, or collect the data, if desired. This means you would write your functionality as an extension, rather than a pull request. In other words, the stats functionality should be written as a plugin that extends the Update Manager plugin.

I took a quick scan of the code and note a couple of issues:

1. Your database queries are not safe for execution. All raw queries should pass through the `prepare()` method without exception. https://developer.wordpress.org/reference/classes/wpdb/prepare/

1. Your `is_safe_url()` method is only checking if the variable is set, rather than checking it the URL is safe.
